GHC 8.10.1 build issues

Currently, `language-docker-0.3.6.0` does not build GHC 8.10.1. This issue aims to document the underlying issues.

1. The first issue you run into is this error:

   <details>

   ```
   [ 7 of 10] Compiling Language.Docker.Syntax.Lift ( src/Language/Docker/Syntax/Lift.hs, interpreted )
   
   src/Language/Docker/Syntax/Lift.hs:16:1: error:
       Illegal kind signature: ‘a_a8rK’
         Perhaps you intended to use KindSignatures
       In an instance declaration
      |
   16 | deriveLift ''Fixed
      | ^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:16:1: error:
       Unexpected kind variable ‘k_a8rJ’
       Perhaps you intended to use PolyKinds
       In an instance declaration
      |
   16 | deriveLift ''Fixed
      | ^^^^^^^^^^^^^^^^^^
   ```
   </details>

   This is because [`Fixed` is poly-kinded as of `base-4.14`](https://gitlab.haskell.org/ghc/ghc/issues/10055). A simple fix is to enable `PolyKinds` at the top of `Language.Docker.Syntax.Lift`.
2. The next issue involves a missing `Lift Text` instance:

   <details>
   
   ```
   [ 7 of 10] Compiling Language.Docker.Syntax.Lift ( src/Language/Docker/Syntax/Lift.hs, interpreted )
   
   src/Language/Docker/Syntax/Lift.hs:23:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_akQJ)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "PortStr"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_akQJ)
         In a case alternative:
             PortStr x0_akQJ
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "PortStr"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_akQJ)
      |
   23 | deriveLift ''Port
      | ^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:27:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_akSf)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "Registry"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_akSf)
         In a case alternative:
             Registry x0_akSf
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "Registry"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_akSf)
      |
   27 | deriveLift ''Registry
      | ^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:29:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x1_akSS)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              ((Language.Haskell.TH.Lib.Internal.appE
                  (Language.Haskell.TH.Lib.Internal.conE
                     ((Language.Haskell.TH.Syntax.Name
                         (Language.Haskell.TH.Syntax.mkOccName "Image"))
                        (((Language.Haskell.TH.Syntax.NameG
                             Language.Haskell.TH.Syntax.DataName)
                            (Language.Haskell.TH.Syntax.mkPkgName
                               "language-docker-8.1.0-inplace"))
                           (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                 (lift x0_akSR)))
             (lift x1_akSS)
         In a case alternative:
             Image x0_akSR x1_akSS
               -> (Language.Haskell.TH.Lib.Internal.appE
                     ((Language.Haskell.TH.Lib.Internal.appE
                         (Language.Haskell.TH.Lib.Internal.conE
                            ((Language.Haskell.TH.Syntax.Name
                                (Language.Haskell.TH.Syntax.mkOccName "Image"))
                               (((Language.Haskell.TH.Syntax.NameG
                                    Language.Haskell.TH.Syntax.DataName)
                                   (Language.Haskell.TH.Syntax.mkPkgName
                                      "language-docker-8.1.0-inplace"))
                                  (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                        (lift x0_akSR)))
                    (lift x1_akSS)
      |
   29 | deriveLift ''Image
      | ^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:31:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_akTx)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "ImageAlias"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_akTx)
         In a case alternative:
             ImageAlias x0_akTx
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "ImageAlias"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_akTx)
      |
   31 | deriveLift ''ImageAlias
      | ^^^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:33:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_akU9)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "Tag"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_akU9)
         In a case alternative:
             Tag x0_akU9
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "Tag"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_akU9)
      |
   33 | deriveLift ''Tag
      | ^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:35:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_akUL)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "Digest"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_akUL)
         In a case alternative:
             Digest x0_akUL
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "Digest"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_akUL)
      |
   35 | deriveLift ''Digest
      | ^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:37:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x4_akVr)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              ((Language.Haskell.TH.Lib.Internal.appE
                  ((Language.Haskell.TH.Lib.Internal.appE
                      ((Language.Haskell.TH.Lib.Internal.appE
                          ((Language.Haskell.TH.Lib.Internal.appE
                              (Language.Haskell.TH.Lib.Internal.conE
                                 ((Language.Haskell.TH.Syntax.Name
                                     (Language.Haskell.TH.Syntax.mkOccName "BaseImage"))
                                    (((Language.Haskell.TH.Syntax.NameG
                                         Language.Haskell.TH.Syntax.DataName)
                                        (Language.Haskell.TH.Syntax.mkPkgName
                                           "language-docker-8.1.0-inplace"))
                                       (Language.Haskell.TH.Syntax.mkModName
                                          "Language.Docker.Syntax")))))
                             (lift x0_akVn)))
                         (lift x1_akVo)))
                     (lift x2_akVp)))
                 (lift x3_akVq)))
             (lift x4_akVr)
         In a case alternative:
             BaseImage x0_akVn x1_akVo x2_akVp x3_akVq x4_akVr
               -> (Language.Haskell.TH.Lib.Internal.appE
                     ((Language.Haskell.TH.Lib.Internal.appE
                         ((Language.Haskell.TH.Lib.Internal.appE
                             ((Language.Haskell.TH.Lib.Internal.appE
                                 ((Language.Haskell.TH.Lib.Internal.appE
                                     (Language.Haskell.TH.Lib.Internal.conE
                                        ((Language.Haskell.TH.Syntax.Name
                                            (Language.Haskell.TH.Syntax.mkOccName "BaseImage"))
                                           (((Language.Haskell.TH.Syntax.NameG
                                                Language.Haskell.TH.Syntax.DataName)
                                               (Language.Haskell.TH.Syntax.mkPkgName
                                                  "language-docker-8.1.0-inplace"))
                                              (Language.Haskell.TH.Syntax.mkModName
                                                 "Language.Docker.Syntax")))))
                                    (lift x0_akVn)))
                                (lift x1_akVo)))
                            (lift x2_akVp)))
                        (lift x3_akVq)))
                    (lift x4_akVr)
      |
   37 | deriveLift ''BaseImage
      | ^^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:41:1: error:
       • Could not deduce (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
         from the context: Lift args
           bound by the instance declaration
           at src/Language/Docker/Syntax/Lift.hs:41:1-24
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_akYg)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "User"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_akYg)
         In a case alternative:
             User x0_akYg
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "User"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_akYg)
      |
   41 | deriveLift ''Instruction
      | ^^^^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:43:1: error:
       • Could not deduce (Lift Filename) arising from a use of ‘lift’
         from the context: Lift args
           bound by the instance declaration
           at src/Language/Docker/Syntax/Lift.hs:43:1-27
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x1_al0d)’
         In the first argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘((Language.Haskell.TH.Lib.Internal.appE
                ((Language.Haskell.TH.Lib.Internal.appE
                    (Language.Haskell.TH.Lib.Internal.conE
                       ((Language.Haskell.TH.Syntax.Name
                           (Language.Haskell.TH.Syntax.mkOccName "InstructionPos"))
                          (((Language.Haskell.TH.Syntax.NameG
                               Language.Haskell.TH.Syntax.DataName)
                              (Language.Haskell.TH.Syntax.mkPkgName
                                 "language-docker-8.1.0-inplace"))
                             (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                   (lift x0_al0c)))
               (lift x1_al0d))’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              ((Language.Haskell.TH.Lib.Internal.appE
                  ((Language.Haskell.TH.Lib.Internal.appE
                      (Language.Haskell.TH.Lib.Internal.conE
                         ((Language.Haskell.TH.Syntax.Name
                             (Language.Haskell.TH.Syntax.mkOccName "InstructionPos"))
                            (((Language.Haskell.TH.Syntax.NameG
                                 Language.Haskell.TH.Syntax.DataName)
                                (Language.Haskell.TH.Syntax.mkPkgName
                                   "language-docker-8.1.0-inplace"))
                               (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                     (lift x0_al0c)))
                 (lift x1_al0d)))
             (lift x2_al0e)
      |
   43 | deriveLift ''InstructionPos
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:45:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_al12)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "SourcePath"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_al12)
         In a case alternative:
             SourcePath x0_al12
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "SourcePath"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_al12)
      |
   45 | deriveLift ''SourcePath
      | ^^^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:47:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_al1E)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "TargetPath"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_al1E)
         In a case alternative:
             TargetPath x0_al1E
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "TargetPath"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_al1E)
      |
   47 | deriveLift ''TargetPath
      | ^^^^^^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:49:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_al2g)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "Chown"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_al2g)
         In a case alternative:
             Chown x0_al2g
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "Chown"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_al2g)
      |
   49 | deriveLift ''Chown
      | ^^^^^^^^^^^^^^^^^^
   
   src/Language/Docker/Syntax/Lift.hs:51:1: error:
       • No instance for (Lift Data.Text.Internal.Text)
           arising from a use of ‘lift’
       • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
           ‘(lift x0_al2S)’
         In the expression:
           (Language.Haskell.TH.Lib.Internal.appE
              (Language.Haskell.TH.Lib.Internal.conE
                 ((Language.Haskell.TH.Syntax.Name
                     (Language.Haskell.TH.Syntax.mkOccName "CopySource"))
                    (((Language.Haskell.TH.Syntax.NameG
                         Language.Haskell.TH.Syntax.DataName)
                        (Language.Haskell.TH.Syntax.mkPkgName
                           "language-docker-8.1.0-inplace"))
                       (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
             (lift x0_al2S)
         In a case alternative:
             CopySource x0_al2S
               -> (Language.Haskell.TH.Lib.Internal.appE
                     (Language.Haskell.TH.Lib.Internal.conE
                        ((Language.Haskell.TH.Syntax.Name
                            (Language.Haskell.TH.Syntax.mkOccName "CopySource"))
                           (((Language.Haskell.TH.Syntax.NameG
                                Language.Haskell.TH.Syntax.DataName)
                               (Language.Haskell.TH.Syntax.mkPkgName
                                  "language-docker-8.1.0-inplace"))
                              (Language.Haskell.TH.Syntax.mkModName "Language.Docker.Syntax")))))
                    (lift x0_al2S)
      |
   51 | deriveLift ''CopySource
      | ^^^^^^^^^^^^^^^^^^^^^^^
   ```
   </details>

   #50 incorrectly attempted to fix this by limiting the build plan to only GHC 8.8 or later. However, that is insufficient, as GHC 8.10.1 bundles `text-1.2.3.2`, which does not define a `Lift Text` instance. (See [here](https://gitlab.haskell.org/ghc/ghc/issues/17956) for more discussion on this.)

   I think the best way to solve this problem would be to depend on `th-lift-instances`, which backports the `Lift Text` instance to older versions of `text`. Since `language-dockerfile` already depends on this library, this doesn't seem like that radical of a change. If we did this, then `language-docker` could even theoretically support older versions of GHC again, although I will leave that choice to the maintainers of this library.

--- a/package.yaml
+++ b/package.yaml
@@ -33,6 +33,7 @@ dependencies:
   - mtl
   - template-haskell
   - th-lift
+  - th-lift-instances
   - text
   - time
   - containers

--- a/src/Language/Docker/Syntax/Lift.hs
+++ b/src/Language/Docker/Syntax/Lift.hs
@@ -1,17 +1,15 @@
+{-# LANGUAGE PolyKinds #-}
 {-# LANGUAGE TemplateHaskell #-}
 
 module Language.Docker.Syntax.Lift where
 
-import qualified Data.ByteString as ByteString
 import Data.Fixed (Fixed)
 import Data.Time.Clock (DiffTime)
+import Instances.TH.Lift () -- Defines Lift instances for ByteString and Text
 import Language.Haskell.TH.Lift
 import Language.Haskell.TH.Syntax ()
 
 import Language.Docker.Syntax
-
-instance Lift ByteString.ByteString where
-    lift b = [|ByteString.pack $(lift $ ByteString.unpack b)|]
 
 deriveLift ''Fixed
 

